src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1084,6 +1084,34 @@ class MacroAssembler: public Assembler {
 public:
   // Calls
 
+  // Trampoline_call generates a call of the provided address.
+  // The address must be inside the code cache.
+  // Supported entry.rspec():
+  // - relocInfo::runtime_call_type
+  // - relocInfo::opt_virtual_call_type
+  // - relocInfo::static_call_type
+  // - relocInfo::virtual_call_type
+  //
+  // If the distance to the address can exceed the branch range
+  // (128M for the release build, 2M for the debug build; see branch_range definition)
+  // for direct calls(BL), a special code with BR (trampoline) is put in the stub code section.
+  // The call is redirected to it. When CodeBuffer is copied to CodeCache, the distance to
+  // callee's address is checked to bypass the trampoline by replacing the call of the trampoline
+  // with the call of the target.
+  //
+  // Trampoline_call is most suitable for calls of Java methods. Java calls callees can be changed
+  // to the interpreter or different versions of a compiled method. Those callees can be
+  // near-distant or far-distant. Trampoline_call supports switching between near-distant callees
+  // and far-distant callees by having a reserved trampoline. The trampoline is only used if needed.
+  //
+  // The code for runtime calls can also be generated with far_call. For possible far-distant callees
+  // far_call does not use the stub code section for additional code. It inserts the code at a call site.
+  // This prevents the call from optimization to a direct call when the code is copied to CodeCache.
+  //
```

Comment:
I can't understand any of this. As far as I can tell it is nonsense.
It introduces pointless confusing terminology such as "far-distant" and "near-distant".
"This prevents the call from optimization to a direct cal" is completely wrongl

Reply:
> "This prevents the call from optimization to a direct cal" is completely wrongl

If I am wrong, could you please point me at the place where 'adrp, add, bl' is optimized?

> As far as I can tell it is nonsense.

Could you please be more constructive?

> It introduces pointless confusing terminology such as "far-distant" and "near-distant".

Please, offer your variant. There is no standard terminology:
- GCC uses: long/short https://gcc.gnu.org/onlinedocs/gcc-8.1.0/gcc/ARM-Function-Attributes.html
- TI uses: relative/absolute/far away/near call https://downloads.ti.com/docs/esd/SPNU118/generate-far-call-trampolines-trampolines-option-stdz0755016.html
- ARM always calls tramopolines veneers: https://developer.arm.com/documentation/dui0803/d/pge1406301797482

We can use: short/long like GCC. What do you think?